Makefile

```diff
@@ -1,144 +1,63 @@
+default: all
 
-.PHONY: run
-run: SHELL:=/usr/bin/env bash
-run: export ADDITIONAL_ACTIVE_PROFILES="local-overrides"
-run: export APPLICATION_NAME=${APP_NAME}
-run: export APPLICATION_VERSION=${VERSION}
-run: build-dirs run-before-tools
-	@echo "Building ${DIST_DIR}/${APP_NAME}..."
-	@go build -ldflags "-X main.version=${VERSION}" -o ${DIST_DIR}/${APP_NAME} main.go
-	if [[ -f "${HOME}/.armory/${APP_NAME}.env" ]] ; then echo "sourcing ~/.armory/${APP_NAME}.env"; source "${HOME}/.armory/${APP_NAME}.env" ; fi
-	${DIST_DIR}/${APP_NAME}
-	@mkdir -p $(GEN_DIR)
+include ./scripts/common_targets.mk
 
+.PHONY: all
+all: clean build-dirs run-before-tools build check run-after-tools
 
-#####################
-## Run tests and generate coverage report
-#####################
-.PHONY: check
-check: export APP_NAME:=$(APP_NAME)
-check: export BUILD_DIR:=$(BUILD_DIR)
-check: build-dirs 
-	@go run scripts/check.go
-
-#####################
-## install all tools step
-#####################
-.PHONY: install-tools
-install-tools:
-	@test ! -f "$(GEN_DIR)/.tools" && \
+.PHONY: tools
+tools:
 	echo installing tools.... && \
 	go install github.com/vakenbolt/go-test-report@v0.9.3 && \
 	go install github.com/undoio/delve/cmd/dlv@latest && \
 	echo installing static check... && \
-	go install honnef.co/go/tools/cmd/staticcheck@latest && \
-	mkdir -p $(GEN_DIR) && touch $(GEN_DIR)/.tools || echo tools are installed
-
-
-#####################
-## Local static-check
-#####################
-.PHONY: static-check
-static-check: install-tools
-	@echo "Running static check in ${PWD}..."
-	@staticcheck ./...
+	go install honnef.co/go/tools/cmd/staticcheck@latest
 
 
-.PHONY: run-before-tools
-run-before-tools: configureGoEnvironment
-run-before-tools: install-tools  
-	@go mod tidy
-	@echo run before tools DONE
-
-.PHONY: run-after-tools
-run-after-tools: static-check
-	@echo run after tools DONE
-
-#####################
-## Custom build steps
-#####################
-.PHONY: configureGoEnvironment
-configureGoEnvironment: 
+.PHONY: configure-build
+configure-build:
 	@go env -w CGO_ENABLED=0
 	@go env
+
 .PHONY: integration
 integration: build-dirs install-tools
 	@go test -v -cover ./integration/... -json > ${BUILD_DIR}/reports/integration-test-report.json
 	@go test -v -coverprofile=${BUILD_DIR}/reports/integration.cov ./integration/...
 	@cat ${BUILD_DIR}/reports/integration-test-report.json | go-test-report --title ${APP_NAME}-integration-test -v --output ${BUILD_DIR}/reports/integration_test_report.html
+
 .PHONY: release
-release: configureGoEnvironment build
+release: clean build-linux-amd64
 	@echo Release version of armory-cli ${VERSION} created in ${DIST_DIR}/${APP_NAME}${APP_EXT}
+	@docker build \
+	--tag $(REGISTRY)$(REGISTRY_ORG)/$(APP_NAME)-cli:$(IMAGE_TAG) \
+	--tag $(REGISTRY)$(REGISTRY_ORG)/$(APP_NAME)-cli:$(VERSION) \
+	--label "org.opencontainers.image.created=$(TIMESTAMP)" \
```

Comment:
normally, all these labels are added by gh's build docker action - I wanted to keep the generated image backwards compatible, so I'm adding these labels manually - they would be generated for both local build as well for any GH build. 
when building locally, the content of PUSH variable is empty - it is populated when GHA_SHA env variable is present - unless you explicitly set it, it would only push within GH workflows, but you can still work with the image locally.